app/util/job.ts

```diff
@@ -194,11 +191,36 @@ export async function resumeAndSaveJob(
         await workflowStep.save(tx);
       }
     }
-    job.resume();
+    if (jobFnName === 'resume') {
+      job.resume();
+    } else if (jobFnName === 'skipPreview') {
+      job.skipPreview();
+    }
```

Comment:
Throw an error here if `jobFnName` is not 'resume' or 'skipPreview'. Probably should do that first thing in this function. Unless there is a valid use case to call this function without either of those.

Reply:
This comment does not apply if you refactor as described in my other comments.

Reply:
Refactored.